### PR TITLE
Associations PR

### PR DESCRIPTION
**Pull request by kkasaija**
- Added rental, classroom and book classes
- Setup class associations for the respective classes

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
   end
 
   def play_hooky
-    '¯\(ツ)/¯'
+    '¯(ツ)/¯'
+  end
+
+  def classroom=(classroom)
+    @classroom = classroom
+    classroom.students.push(self)
```
Comment (t-yanick) [inline student.rb]: - [ ] Great job writing the method that serves as belongs-to, but if you take a closer look as this [article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships), such relationships have the `include` method. I think you should have something like `classroom.students.push(self) unless classroom.students.include?(self)`

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @Kasaija-Kenneth , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file
:heavy_check_mark: Linters checks passing. 

There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb (hunk 1 above)
Comment (kkasaija) [reply student.rb]: Thanks for the direction @t-yanick

**Review by BenMukebo — APPROVED** [review]
### Hi @Kasaija-Kenneth, 👋 

# STATUS: Approved  🎉 

Great work on making the changes requested by a previous reviewer :1st_place_medal:  
 Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 



### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- No Linter errors :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kkasaija** [comment]
Thank you @BenMukebo  and @t-yanick 
I am slowly but steadily beginning to reason like a programmer 🤝🤝🤝🤝🤝

**Pull request merged**